lib/portfinder.js

```diff
@@ -29,28 +30,48 @@ internals.testPort = function(options, callback) {
 
   options.port   = options.port   || exports.basePort;
   options.host   = options.host   || null;
-  options.server = options.server || net.createServer(function () {
-    //
-    // Create an empty listener for the port testing server.
-    //
+  options.server = options.server || net.createServer(function (connection) {
+    connection.pipe(connection);
   });
 
   debugTestPort("entered testPort(): trying", options.host, "port", options.port);
 
   function onListen () {
+    internals.windows10BindError = {
+      message: 'client and server unable to communicate over '+options.server.address(),
+      code: 'windows10BindError'
+    }
+
     debugTestPort("done w/ testPort(): OK", options.host, "port", options.port);
 
-    options.server.removeListener('error', onError);
-    options.server.close();
-    callback(null, options.port);
+    var client = net.createConnection(options, function() {
+      client.write('OK PORTFINDER');
+    });
+
+    client.setEncoding('utf8');
+    client.on('data', function(data) {
+      debugTryConnection('Try Connection client -> server -> client: ', data);
+
+      if (data === 'OK PORTFINDER') {
+        options.server.removeListener('error', onError);
+        options.server.close();
+        client.end();
+      } else {
+        options.server.emit('error', new Error(internals.windows10BindError));
+      }
+    });
+
+    client.on('end', function() {
+      callback(null, options.port);
+    });
```

Comment:
future portfinder 2.0 partial code - includes fix for windows 10 native shell port rebinding and macbookpro with touchbar - and any other future issues that arise when **listening** on a port is not adequate - to use this portfinder needs to catch any OS errors that occur from trying to pipe data through a protected interface, such a bluetooth(and likely macbookpro touchbar) both which are interfaces returned by `os.networkInterfaces()`